oligo_designer_toolsuite/oligo_specificity_filter/_filter_base.py

```diff
@@ -106,34 +147,45 @@ def apply(
         :type reference_database: ReferenceDatabase
         :return: The filtered oligo database with sequences having significant hits removed.
         :rtype: OligoDatabase
+
         """
         options = get_args(_TYPES_SEQ)
         assert (
             sequence_type in options
         ), f"Sequence type not supported! '{sequence_type}' is not in {options}."
 
         # Create index file for search
-        file_reference = reference_database.write_database_to_fasta(filename="reference_db")
-        filename_reference_index = self._create_index(file_reference=file_reference, n_jobs=n_jobs)
+        file_reference = reference_database.write_database_to_fasta(
+            filename="reference_db"
+        )
+        file_index = self._create_index(
+            file_reference=file_reference, n_jobs=n_jobs
+        )
 
         # Run search for each region in parallel
         region_ids = list(oligo_database.database.keys())
-        results = Parallel(n_jobs=n_jobs)(
+        table_hits = Parallel(n_jobs=n_jobs)(
             delayed(self._run_filter)(
                 sequence_type=sequence_type,
                 region_id=region_id,
                 oligo_database=oligo_database,
-                file_index=filename_reference_index,
+                file_index=file_index,
+                file_reference=file_reference,
                 consider_hits_from_input_region=True,
             )
             for region_id in region_ids
         )
+        # delete the temporary files 
         os.remove(file_reference)
-
-        # Process results
-        table_hits, oligos_with_hits = zip(*results)
-
-        for region_id, oligos_with_hits_region in zip(region_ids, oligos_with_hits):
+        file_index_basename = os.path.basename(file_index)
```

Comment:
Move into a seperate function, e.g. _remove_index(file_index)
and reduce code, e.g.:

```
def _remove_index(self, file_index):
    file_index_basename = os.path.basename(file_index)
    directory = Path(self.dir_output)
    for file_path in directory.glob(file_index_basename + ".*"):
        os.remove(file_path)
```